File: JAVA/Halley.java

//Autor: Henirque Grissi
public class Halley {
        public static void main(String args[]) {
        int ano = MyIO.readInt(); //Recebe o primeiro ano
        int passagem = 0;
        while (ano != 0){
            passagem = ano + 76; //Calcula quandos erá a passagem
            System.out.println(passagem);
            ano = MyIO.readInt(); //Recebe outro ano e testa no loop
        }
    }
}

File: C/Halley.c
//Autor: Henrique Grissi
int main (void){
    int ano, passagem;
    scanf("%d",&ano); //Recebe primeiro ano
    while(ano != 0){ //Condição de parada
        passagem = ano + 76; //Calcula quando será  a passagem
        printf("%d",passagem); 
        scanf("%d",&ano); //Recebe novo ano e tyesta no loop
    }
}